config/configload/validate.go

```diff
@@ -5,68 +5,124 @@ import (
 	"regexp"
 
 	"github.com/hashicorp/hcl/v2"
+
+	"github.com/avenga/couper/config"
 )
 
+var regexLabel = regexp.MustCompile(`^[a-zA-Z0-9_]+$`)
+
 // https://datatracker.ietf.org/doc/html/rfc7231#section-4
 // https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6
 var methodRegExp = regexp.MustCompile("^[!#$%&'*+\\-.\\^_`|~0-9a-zA-Z]+$")
 
-func validLabelName(name string, hr *hcl.Range) error {
-	if !regexProxyRequestLabel.MatchString(name) {
-		return hcl.Diagnostics{&hcl.Diagnostic{
-			Severity: hcl.DiagError,
-			Summary:  "label contains invalid character(s), allowed are 'a-z', 'A-Z', '0-9' and '_'",
-			Subject:  hr,
-		}}
+func newDiagErr(subject *hcl.Range, summary string) error {
+	return hcl.Diagnostics{&hcl.Diagnostic{
+		Severity: hcl.DiagError,
+		Summary:  summary,
+		Subject:  subject,
+	}}
+}
+
+func validLabel(name string, subject *hcl.Range) error {
+	if !regexLabel.MatchString(name) {
+		return newDiagErr(subject, "label contains invalid character(s), allowed are 'a-z', 'A-Z', '0-9' and '_'")
```

Comment:
this message is thrown for empty name, too.

Reply:
I have added an additional check.